include/rocksdb/utilities/object_registry.h

```diff
@@ -190,16 +234,46 @@ class ObjectRegistry {
   }
 
   // Dump the contents of the registry to the logger
-  void Dump(Logger* logger) const;
+  virtual void Dump(Logger* logger) const = 0;
 
- private:
-  const ObjectLibrary::Entry* FindEntry(const std::string& type,
-                                        const std::string& name) const;
+  // Creates a new ObjectLibrary within this ObjectRegistry and invokes
+  // registrar(arg) to register the appropriate factories in the new library.
+  // @param id The name for this library
+  // Returns the number of factories registered.
+  virtual std::shared_ptr<ObjectLibrary> AddProgramLibrary(
+      const std::string& id, const RegistrarFunc& registrar,
+      const std::string& arg) = 0;
+
+  // Creates a new ObjectLibrary within this ObjectRegistry.
+  // Loads the registrar func named "method" from the input library and, upon
+  // success, invokes registrar(arg) to register the appropriate factories in
+  // the new library. Returns OK if method was found in the library and non-OK
+  // otherwise.
+  virtual Status AddLoadedLibrary(
```

Comment:
Is there a way to verify the plugin was built against compatible headers?
Also I wonder if there is any effort to expand our ABI compatibility guarantees -- that should make this approach more compelling. Currently we only do best-effort to uphold ABI compatibility across patch versions within a minor version.

Reply:
In #7949 I propose how to do better ABI validation.  I can remove this method until that is approved if that helps this one move forward.

Reply:
I removed the AddLoadedLibrary method for now.